crates/polars-lazy/src/scan/file_list_reader.rs

```diff
@@ -11,23 +12,124 @@ use crate::prelude::*;
 
 pub type PathIterator = Box<dyn Iterator<Item = PolarsResult<PathBuf>>>;
 
-// cloud_options is used only with async feature
-#[allow(unused_variables)]
-fn polars_glob(pattern: &str, cloud_options: Option<&CloudOptions>) -> PolarsResult<PathIterator> {
-    if is_cloud_url(pattern) {
+fn expand_paths(
```

Comment:
Maybe add a docstring explaining that this does globbing and/or recursive directory traversal.

Reply:
added